AutoRest/Generators/NodeJS/NodeJS/ClientModelExtensions.cs

```diff
@@ -242,6 +242,28 @@ private static string ValidatePrimaryType(this PrimaryType primary, IScopeProvid
             }
         }
 
+        /// <summary>
+        /// Returns the TypeScript type string for the specified primary type
+        /// </summary>
+        /// <param name="primary">primary type to query</param>
+        /// <returns>The TypeScript type correspoinding to this model primary type</returns>
+        private static string PrimaryTSType(this PrimaryType primary) {
+            if (primary == PrimaryType.Boolean)
+                return "boolean";
+            else if (primary == PrimaryType.Double || primary == PrimaryType.Int || primary == PrimaryType.Long)
+                return "number";
+            else if (primary == PrimaryType.String)
+                return "string";
+            else if (primary == PrimaryType.Date || primary == PrimaryType.DateTime)
+                return "Date";
+            else if (primary == PrimaryType.Object)
+                return "Object";   // TODO: test this
```

Comment:
I chatted with @markcowl about this.   PrimaryType.Object is used when there's no schema type specified.   The question is whether primitive types (e.g. string and number) should be allowed here or just dictionaries.  I think it's a little safer to keep it as a TypeScript "Object", which is the more permissive type (and can't hurt), but after the code is merged I will test this more and chat with Amar about his intention for how the JavaScript code should behave in this scenario.